src/qililab/qprogram/qblox_compiler.py

```diff
@@ -380,8 +400,12 @@ def __handle_static_sync(self, buses: set[str]):
         for bus in buses:
             duration_diff = max_duration - self._buses[bus].static_duration
             if duration_diff > 0:
+                # loop over wait instructions if static duration is longer than allowed qblox max wait time of 2**16 -4
+                if duration_diff > INST_MAX_WAIT:
+                    self._handle_long_wait(bus=bus, duration=duration_diff)
+                # add the remaining wait time (or all of it if the above conditional is false)
                 self._buses[bus].qpy_block_stack[-1].append_component(
-                    component=QPyInstructions.Wait(wait_time=duration_diff)
+                    component=QPyInstructions.Wait(wait_time=duration_diff % INST_MAX_WAIT)
                 )
```

Comment:
Could you add all of these under one method? Because now we have half of the logic in `_handle_long_wait` and half in the lines 407-409. And many lines are repeated anyway. We could have everything in one method `__add_wait_instructions`.